### PR TITLE
Select Account to backup

### PR DESCRIPTION
**Pull request by antonijzelinskij**
#8694dmmct

### CODE REVIEW

**Comment by stepanLav** [comment]
Task linked: [CU-8694dmmct Android. Backup manually — Select key to backup](https://app.clickup.com/t/8694dmmct)

File: feature-account-impl/src/main/java/io/novafoundation/nova/feature_account_impl/domain/manualBackup/ManualBackupInteractor.kt [hunk 1]
```diff
@@ -8,20 +8,55 @@ import io.novafoundation.nova.feature_account_api.domain.model.LightMetaAccount.
...
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+
+class MetaAccountChains(
+    val metaAccount: MetaAccount,
+    val defaultChains: List<Chain>,
+    val customChains: List<Chain>
+)
 
 interface ManualBackupInteractor {
```
Comment (valentunn) [inline feature-account-impl/src/main/java/io/novafoundation/nova/feature_account_impl/domain/manualBackup/ManualBackupInteractor.kt:24]: Not sure we need to put everything into single interactor. I suggest to create one interactor per screen

File: feature-account-impl/src/main/java/io/novafoundation/nova/feature_account_impl/presentation/manualBackup/accounts/ManualBackupSelectAccountViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+    fun backClicked() {
+        router.back()
+    }
+
+    private fun mapMetaAccountToUI(
+        metaAccountChains: MetaAccountChains
+    ): List<ManualBackupRVItem> {
+        return buildList {
+            if (metaAccountChains.defaultChains.isNotEmpty()) {
```
Comment (valentunn) [inline feature-account-impl/src/main/java/io/novafoundation/nova/feature_account_impl/presentation/manualBackup/accounts/ManualBackupSelectAccountViewModel.kt:47]: Iirc we have similar logic in Wallet Connect Session Approve screen. Can you please check whether we can reuse it or not?

**Review by valentunn — APPROVED** [review]

File: feature-account-impl/src/main/java/io/novafoundation/nova/feature_account_impl/presentation/manualBackup/accounts/ManualBackupSelectAccountViewModel.kt (hunk 2 above)
Comment (antonijzelinskij) [reply feature-account-impl/src/main/java/io/novafoundation/nova/feature_account_impl/presentation/manualBackup/accounts/ManualBackupSelectAccountViewModel.kt:47]: I checked it and think this logic is not very reusable in our case and also my logic is pretty simple and I think we can left it like that

**Review by valentunn — APPROVED** [review]

**Pull request merged**